core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -638,9 +630,15 @@ angular.module('oppia').directive('questionsList', [
             });
           };
 
-          $scope.$on(EVENT_QUESTION_SUMMARIES_INITIALIZED, _initTab);
+          $scope.$on(EVENT_QUESTION_SUMMARIES_INITIALIZED, function(ev) {
+            _initTab(false);
+          });
+          $scope.$on('changeSelectedSkill', function(ev, skillId) {
```

Comment:
Let's call this onSelectedSkillChanged.

Better still, though -- can we not use an event at all? Instead, pass the skill ID through the directive as an attribute. Events are like gotos, they can make things quite hard to maintain.

Reply:
I was doing that, but we need to fetch the questions as the skills are selected. So, a function needed to be invoked as the options in dropdowns are selected. That's why I used broadcasts.

Reply:
Renamed the broadcast variable name.

Reply:
Sorry, I don't quite understand this. I understand that a function needs to be invoked when the option in the dropdown is selected.

But why does that mean you need to use a broadcast? Couldn't you have a QuestionCacheService which is called (and fetches questions for the frontend) when the dropdown option is selected? The question tab can then use this to retrieve the list of questions when the user navigates to that tab.

Reply:
Ah, ok. Done, used QuestionsListService itself to take care of pagination and caching. The caching was already done by the service, but it wasn't utilised in this way.